controllers/servers/csi/decorators.py

```diff
@@ -50,3 +53,14 @@ def _set_sync_lock(lock_id, lock_request_attribute, error_response_type,
         return handle_exception(ex, context, grpc.StatusCode.ABORTED, error_response_type)
     logger.info("finished {}".format(controller_method_name))
     return response
+
+
+def register_csi_plugin():

```

Comment:
have you considered adding the UNIQUE_KEY as an argument to this decorator?

Reply:
no, because it gets the unique key from extracting it from the registration_map using the function name as a key

Reply:
I know, so now you have to keep the registration array separate from the function definition. Passing the unique key as the decorator argument could keep the code in one place and obvious to a reader who doesn't know about the `REGISTRATION_MAP`.

It would be

```   
    @register_csi_plugin('volume_group')
    def remove_volume_from_volume_group(self, volume_id):
```

and no need to keep the separate file of `registration_maps.py`, as you'll capture the values inside the decorator. The below wasn't tested, but something along the lines:

```
def register_csi_plugin(unique_key, metadata=''):
    def wrapper():
          @decorator
          def call_csi_plugin_registration(mediator_method, mediator_class, *args):
              mediator_class.register_plugin(unique_key, metadata)
              return mediator_method(mediator_class, *args)
      
          return call_csi_plugin_registration
    return wrapper
```

Reply:
I see some advantages and disadvantages in this approach as the approach that I took, I don't think that it will change much, so I can leave with both ways

Reply:
what are the disadvantages?

Reply:
I think it can require more maintenance when you want to change a unique key because you will change it in multiple places, and I believe using registration maps will let us see which functions belong to which unique keys

Reply:
If someone decides to rename a handler method they will have to know that they need to go and also rename the key in the registration map, which is not obvious, and that's my main concern. When you have the unique id configured in the decorator to a handler, that's the only place you need to change it if you want to, and it's obvious which handler also reports a particular plugin usage.

If you're worried about multiple different handlers using the same unique id, you can surely define constants and use those instead of a literal, like

```
VOLUME_GROUP_FEATURE='volume_group'
...

 @register_csi_plugin(VOLUME_GROUP_FEATURE)
    def remove_volume_from_volume_group(self, volume_id):
...

 @register_csi_plugin(VOLUME_GROUP_FEATURE)
    def another_handler(self, volume_id):
```